teps/0091-trusted-resources.md

```diff
@@ -364,26 +373,70 @@ type KeylessRef struct {
 
 ### Configuration
 
-Reconciler will respond to verification failure based on ConfigMap settings to:
-1) Return error in reconciler, mark the status as error with `FailedVerification` reason;
-2) Not return error, but mark the TaskRun/PipelineRun as `FailedVerification`;
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  no-match-policy: "deny"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`no-match-policy`. (Optional, `allow`, `deny` or `warn`, default `allow`):
+ * `deny`: Reject the resource and fail the taskrun/pipelinerun if no matching policies are found.
+ * `allow`: Allow the resource if no matching policies are found.
+ * `warn`: Allow the resource and log a warning if no matching policies are found.
+
+ Reconciler will respond to verification failure based on ConfigMap and Verification Policy exsitence settings to:
+1) Return error in reconciler, mark the status as error with `FailedVerification` reason if `enforce` policies fail
+2) Not return error, but warn the log and update the status condition with `FailedVerification` reason if only found `warn` policies.
+3) Skip the verification if no policies exist and `no-match-policy` is set to `allow`.
+
+ **Note:**
+ * If the `no-match-policy` is set to `deny`, resources that do not match any policies will be rejected. Resources that match policies will be accepted if all the matching `enforce` policies pass or only match `warn` policies.
```

Comment:
when you say "rejected" do you mean they can't be written to the cluster (enforced by a webhook when applying a Task/Pipeline), or that TaskRuns/PipelineRuns that use them will fail?

Reply:
Thanks! I think I need to clear about the wording and make clarifications. 
I think ideally it should mean both. If a task/pipeline fails verification, it should not be written into the cluster and TaskRuns/PipelineRuns that use them will fail. 

For remote resources, they are resolved by remote resolution then verified by trusted resources and will not be written into storage. 
For api resources, we can add verification at webhook to reject them

Reply:
Updated related words